### PR TITLE
Localization test flow

### PR DESCRIPTION
**Pull request by akirk**
Run with: 
`./node_modules/mocha/bin/mocha specs-i18n/search-wordpress-on-google.js lib/after.js`

Compare with our Localisation test instructions (Team Global meetup project):

### WordPress.com Signup Flow Test Script
Please take notes & screenshots as you go through the steps so you can prepare for the final report.

1. Set your browser to the locale of your choice. Log out from your WordPress.com account.
2. Go to the most popular search engine for the locale (e.g. https://www.google.co.jp/ for Japanese) and search for “WordPress”.
   - Does the search ad for WordPress.com look ok? Does it sound natural and enticing?
   - Does locale.wordpress.com show up on the first page? How does the description sound?
   - Are there any untranslated WordPress.com pages indexed?
   - (Bonus) Do you find any ads or site listings that violates the trademark/domain policy?
3. Click the locale.wordpress.com link (not the ad) and go to the localized home page. Click around and see if you can find any issues on this page or any other linked pages.
4. Create a new dummy account (create a Gmail Plus address if you need an additional address) & go through the signup process.
   - Check the signup process pages for any issues using a popular local payment method (credit/debit/prepaid card, bank wire via paypal) rather than WPCOM credits (don’t forget to cancel & get a refund later).
   - Are all email messages correctly translated? Do all the links in the email work?

### CODE REVIEW

File: lib/pages/google-search.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+var webdriver = require( 'selenium-webdriver' );
```
Comment (yoavf) [inline lib/pages/google-search.js]: Move this to lib/pages/external ?

Comment (akirk) [inline lib/pages/google-search.js]: Makes sense, I'll do that.

File: lib/pages/external/google-search.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+import BaseContainer from '../../base-container.js';
+
+const by = webdriver.By;
+
+export default class GoogleSearchPage extends BaseContainer {
+	constructor( driver, { location = false, culture = 'en', googleDomain = 'com' } = {} ) {
+		var url = "https://adwords.google.com/apt/anon/AdPreview";
+		super( driver, by.id( 'com.google.ads.apps.anonymousapt.GwtModule' ), true, url );
+		this.waitForPage();
+		this.driver.sleep( 2000 );
```
Comment (akirk) [inline lib/pages/external/google-search.js]: I feel like there must be a more elegant way to wait for the elements to load?

Comment (yoavf) [inline lib/pages/external/google-search.js]: `this.driver.wait( until.elementLocated( this.expectedElementSelector )` ?

**Comment by akirk** [comment]
Example output:

```$ ./node_modules/mocha/bin/mocha specs-i18n/search-wordpress-on-google.js lib/after.js
  Search for WordPress on Google
https://wordpress.com/com-vs-org/

$ CULTURE=de LOCATION=Vienna,\ Austria ./node_modules/mocha/bin/mocha specs-i18n/search-wordpress-on-google.js lib/after.js
  Search for WordPress on Google
https://de.wordpress.com/create/

$ CULTURE=pt-br LOCATION=Sao\ Paulo ./node_modules/mocha/bin/mocha specs-i18n/search-wordpress-on-google.js lib/after.js
  Search for WordPress on Google
https://br.wordpress.com/create/
```

**Comment by akirk** [comment]
```$ CULTURE=it LOCATION=Rome,\ Italy ./node_modules/mocha/bin/mocha specs-i18n/search-wordpress-on-google.js lib/after.js


  Search for WordPress on Google
https://it.wordpress.com/create/
wordpress.com - Crea un Sito WordPress®
www.wordpress.com/
Crea un Blog o un Sito Web Premium. Supporto Live Superiore. Inizia!
```

**Comment by akirk** [comment]
Ouch: Arabic ad for Hebrew search.
```
$ CULTURE=he LOCATION=Jerusalem ./node_modules/mocha/bin/mocha specs-i18n/search-wordpress-on-google.js lib/after.js
  Search for WordPress on Google
https://ar.wordpress.com/create/
WordPress.com®إنشاء موقع
www.wordpress.com/
إنشاء مدونة أو موقع على الإنترنت. مساعدة متفوقة. ابدء الان!
    ✓ Can load Google (8354ms)
```

**Comment by akirk** [comment]
Current flow in action:
google-search

**Comment by akirk** [comment]
We now navigate on to the landing page
>  Search for WordPress on Google
>    ✓ Google search contains our ad (8261ms)
>    ✓ Our landing page exists (2225ms)

**Comment by akirk** [comment]
Now we can run a test like this: `LOCALE_TEST=de-wordpress ./node_modules/mocha/bin/mocha specs-i18n/search-wordpress-on-google.js lib/after.js`

**Comment by hewsut** [comment]
Updated way to run a test: LOCALE_TEST=de ./node_modules/mocha/bin/mocha specs-i18n/search-wordpress-on-google.js lib/after.js

**Comment by deBhal** [comment]
I've rebased to fix conflicts with master, but running the tests locally seems to be failing. The ads are visible, but it looks like we load a new page and then fail.

I've confirmed that the problem with the tests was there before that (https://github.com/Automattic/wp-e2e-tests-i18n has pinned the old SHA1), so I've gone ahead and pushed the rebase.

I rebuilt https://github.com/Automattic/wp-e2e-tests-i18n, and it successfully finds the ads for the `de` and `fr`, but both fail running locally with the same `wp-e2e-tests`. Could be a timing thing, or I might not have the environment right locally.

**Comment by lancewillett** [comment]
@deBhal Could `https://github.com/Automattic/wp-e2e-tests-i18n` be made public? It looks private (and this repo is open to the public).

**Comment by hoverduck** [comment]
@akirk / @yoavf - Is there any reason this was never merged?  I was getting the wp-e2e-tests-i18n wrapper repo set up to handle the screenshot generation (#494) tests and notice that it's set to only run against this specific commit.

It looks like there are a fair number of conflicts now that would need to be resolved before merging, but I can help out with that if necessary.

cc: @rachelmcr

**Comment by akirk** [comment]
@hoverduck I've been on leave for the past months but happy to bring up to date and merge it when I'm back (maybe even earlier). Sorry for the loose ends! If you want to get it moving a bit faster, feel free to commit to the PR.

**Comment by hoverduck** [comment]
@akirk Thanks for the quick reply.  As long as you didn't have a set reason for avoiding merging I'll go ahead and work on rebasing it and get it taken care of.

**Comment by hoverduck** [comment]
Just FYI, I'm working on these updates and hitting a lot of issues integrating some of the changes from #504 and #494 (multiple PRs tied to that issue).  Some of the changes are potentially breaking, so I created a new branch off of this one to help keep things straight in my mind and not have to rewind any commits since I know this branch works as-is at the moment.

https://github.com/Automattic/wp-e2e-tests/tree/team-global-l10n-test-srs

**Comment by akirk** [comment]
@hoverduck great, so we should then probably close this PR and create a new one for the new branch.

**Comment by hoverduck** [comment]
Closed in favor of #635

**Pull request closed without merging**